### PR TITLE
Generate valid Main.lhs files with `cabal init`.

### PR DESCRIPTION
**Pull request by hrhino**
Fixes haskell/cabal#5577.

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [x] Any changes that could be relevant to users have been recorded in the changelog.
* [ ] The documentation has been updated, if necessary.
* [ ] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

> Please also shortly describe how you tested your change. Bonus points for added tests!

I have not. Pointers on how to do so are very appreciated!

### CODE REVIEW

File: cabal-install/Distribution/Client/Init.hs [hunk 1]
```diff
@@ -839,14 +839,23 @@ hasMainHs flags = case mainIs flags of
...
   , "main :: IO ()"
   , "main = putStrLn \"Hello, Haskell!\""
   ]
+  where
+    prependPrefix "" = ""
+    prependPrefix line
+      | isLiterate = "> " ++ line
+      | otherwise  = line
+    isLiterate = case mainIs flags of
+      Flag mainPath -> takeExtension mainPath == ".lhs"
```
Comment (hrhino) [inline cabal-install/Distribution/Client/Init.hs]: `"lhs" isExtensionOf mainPath`

File: cabal-install/Distribution/Client/Init.hs [hunk 2]
```diff
@@ -839,14 +839,23 @@ hasMainHs flags = case mainIs flags of
              || packageType flags == Flag LibraryAndExecutable)
   _ -> False
 
--- | Default Main.hs file.  Used when no Main.hs exists.
-mainHs :: String
-mainHs = unlines
+-- | Default Main.(l)hs file.  Used when no Main.(l)hs exists.
+mainHs :: InitFlags -> String
+mainHs flags = (unlines . fmap prependPrefix)
```
Comment (23Skidoo) [inline cabal-install/Distribution/Client/Init.hs]: I'd use `map` instead of `fmap` here.

**Comment by 23Skidoo** [comment]
Merged, thanks!

**Comment by hrhino** [comment]
Thank _you_! I'll be back...

**Pull request merged**